src/main/java/com/thymeleaf/course/GreetingController.java

```diff
@@ -0,0 +1,37 @@
+package hello;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import javax.validation.Valid;
+
+@Controller
+public class GreetingController {
+
+    private User currentUser=new User();
```

Comment:
If i understand, this object shouldn't be overridden. Make it final. And you can use constructor instead, to initialize object.